src/base.rs

```diff
@@ -322,14 +322,22 @@ unsafe fn members(groups: *mut *mut c_char) -> Vec<OsString> {
 /// ```
 pub fn get_user_by_uid(uid: uid_t) -> Option<User> {
     let mut passwd = unsafe { mem::zeroed::<c_passwd>() };
-    let mut buf = vec![0; 2048];  // TODO: Retry with larger buffer sizes
+    let mut buf = vec![0; 2048];
     let mut result = ptr::null_mut::<c_passwd>();
 
     #[cfg(feature = "logging")]
     debug!("Running getpwuid_r for user #{}", uid);
 
-    unsafe {
-        libc::getpwuid_r(uid, &mut passwd, buf.as_mut_ptr(), buf.len(), &mut result);
+    loop {
+        let r = unsafe {
+            libc::getpwuid_r(uid, &mut passwd, buf.as_mut_ptr(), buf.len(), &mut result)
+        };
+
+        if r != libc::ERANGE {
+            break;
+        }
+
+        buf.resize(buf.len() * 2, 0);
```

Comment:
These lines are causing borrow checker errors on 1.31. Defining the new length on the line above should help the old compiler out.

Reply:
Ta for the reminder, fixed.